lib/cartopy/crs.py

```diff
@@ -1778,6 +1778,80 @@ def y_limits(self):
         return self._y_limits
 
 
+class Sinusoidal(Projection):
+    """
+    A Sinusoidal projection.
+
+    This projection is equal-area.
+    """
+
+    def __init__(self, central_longitude=0.0, false_easting=0.0,
+                 false_northing=0.0, globe=None):
+        """
+        Kwargs:
+
+            * central_longitude - The central longitude. Defaults to 0.
+            * false_easting - X offset from planar origin in metres.
+                              Defaults to 0.
+            * false_northing - Y offset from planar origin in metres.
+                               Defaults to 0.
+            * globe - A :class:`cartopy.crs.Globe`.
+                      If omitted, a default globe is created.
+        """
+        proj4_params = [('proj', 'sinu'),
+                        ('lon_0', central_longitude),
+                        ('x_0', false_easting),
+                        ('y_0', false_northing)]
+        super(Sinusoidal, self).__init__(proj4_params, globe=globe)
+
+        # Obtain boundary points
+        points = []
+        n = 91
+        geodetic_crs = self.as_geodetic()
+        for lat in np.linspace(-90, 90, n):
+            points.append(
+                self.transform_point(180 + central_longitude,
+                                     lat, geodetic_crs)
+            )
+        for lat in np.linspace(90, -90, n):
+            points.append(
+                self.transform_point(-180 + central_longitude,
+                                     lat, geodetic_crs)
+            )
+        points.append(
+            self.transform_point(180 + central_longitude, -90, geodetic_crs))
+
+        self._boundary = sgeom.LineString(points[::-1])
+        x = [p[0] for p in points]
+        y = [p[1] for p in points]
+        self._x_limits = min(x), max(x)
```

Comment:
Right, I'll update this.  This pattern is also found in several other classes in crs.py